client/src/lib/posts/hooks/useSharingPosts.ts

```diff
@@ -0,0 +1,69 @@
+import {useCallback} from 'react';
+import useSessionState from '../../session/state/state';
+import useUserState, {
+  getCurrentUserStateSelector,
+  PostPayload,
+} from '../../user/state/state';
+import {addPost, fetchPosts} from '../api/posts';
+
+const useSharingPosts = (exerciseId?: string) => {
+  const addEvent = useUserState(state => state.addEvent);
+  const events = useUserState(
+    state => getCurrentUserStateSelector(state)?.events,
+  );
+  const session = useSessionState(state => state.asyncSession);
+  const getSharingPosts = useCallback(async () => {
+    if (exerciseId) {
+      return fetchPosts(exerciseId);
+    }
+    return [];
+  }, [exerciseId]);
+
+  const addSharingPost = useCallback(
+    async (
+      sharingId: string,
+      text: string,
+      isPublic: boolean,
+      isAnonymous: boolean,
+    ) => {
+      if (exerciseId && session?.id) {
+        if (isPublic) {
+          await addPost(exerciseId, sharingId, text, isAnonymous);
+        }
+        addEvent({
```

Comment:
Nice 👍

Reply:
Just wondering if `type` and `event` should be passed in to `addEvent`.

And the responsibility to structure in `type`, `payload` and generate a timestamp, would be included in that method.

Reply:
What happened here? Didn't we say to have these as separate arguments?

Reply:
Didn't remember us deciding anything about that, it's all typed anyways. Updated as I believe you mean?